ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/handler/AbortedTxnCleaner.java

```diff
@@ -0,0 +1,168 @@
+        txnHandler.markCleanedForAborts(info);
+        return;
+      }
+      if (MetaStoreUtils.isNoCleanUpSet(t.getParameters())) {
+        // The table was marked no clean up true.
+        LOG.info("Skipping table {} clean up, as NO_CLEANUP set to true", info.getFullTableName());
+        return;
+      }
+      if (!isNull(info.partName)) {
+        p = resolvePartition(info.dbname, info.tableName, info.partName);
+        if (isNull(p)) {
+          // The partition was dropped before we got around to cleaning it.
+          LOG.info("Unable to find partition {}, assuming it was dropped.",
+                  info.getFullPartitionName());
+          txnHandler.markCleanedForAborts(info);
+          return;
+        }
+        if (MetaStoreUtils.isNoCleanUpSet(p.getParameters())) {
+          // The partition was marked no clean up true.
+          LOG.info("Skipping partition {} clean up, as NO_CLEANUP set to true", info.getFullPartitionName());
+          return;
+        }
+      }
+
+      String location = CompactorUtil.resolveStorageDescriptor(t,p).getLocation();
+      info.runAs = TxnUtils.findUserToRunAs(location, t, conf);
+      abortCleanUsingAcidDir(info, location, minOpenTxn);
+
+    } catch (Exception e) {
+      LOG.error("Caught exception when cleaning, unable to complete cleaning of {} due to {}", info,
+              e.getMessage());
+      throw new MetaException(e.getMessage());
+    } finally {
+      if (metricsEnabled) {
+        perfLogger.perfLogEnd(AbortedTxnCleaner.class.getName(), cleanerMetric);
+      }
+    }
+  }
+
+  private void abortCleanUsingAcidDir(AcidTxnInfo info, String location, long minOpenTxn) throws Exception {
+    ValidTxnList validTxnList =
+            TxnUtils.createValidTxnListForAbortedTxnCleaner(txnHandler.getOpenTxns(), minOpenTxn);
+    //save it so that getAcidState() sees it
+    conf.set(ValidTxnList.VALID_TXNS_KEY, validTxnList.writeToString());
+
+    ValidReaderWriteIdList validWriteIdList = getValidCleanerWriteIdList(info, validTxnList);
+
+    // Set the highestWriteId of the cleanup equal to the min(minOpenWriteId - 1, highWatermark).
+    // This is necessary for looking at the complete state of the table till the min open write Id
+    // (if there is an open txn on the table) or the highestWatermark.
+    // This is used later on while deleting the records in TXN_COMPONENTS table.
+    info.highestWriteId = Math.min(isNull(validWriteIdList.getMinOpenWriteId()) ?
+            Long.MAX_VALUE : validWriteIdList.getMinOpenWriteId() - 1, validWriteIdList.getHighWatermark());
+    Table table = metadataCache.computeIfAbsent(info.getFullTableName(), () -> resolveTable(info.dbname, info.tableName));
+    LOG.debug("Cleaning based on writeIdList: {}", validWriteIdList);
+
+    boolean success = cleanAndVerifyObsoleteDirectories(info, location, validWriteIdList, table);
+    if (success || CompactorUtil.isDynPartAbort(table, info.partName)) {
```

Comment:
isDynPartAbort could be removed from CompactionCleaner as it doesn't handle aborts anymore

Reply:
This is part of the cleanup task when the config will be removed. Until then this is supposed to be here.

Reply:
👍